service/src/main/java/com/derpgroup/derpwizard/resource/AlexaResource.java

```diff
@@ -78,41 +89,52 @@ public AlexaResource(MainConfig config, Environment env) {
-    manager.handleRequest(voiceInput, builder);
-
-    SpeechletResponseEnvelope responseEnvelope = new SpeechletResponseEnvelope();
-    responseEnvelope.setSessionAttributes(request.getSession().getAttributes());
-
-    // Create a VoiceOutput object with the SSML content generated by the manater
-    if ((voiceInput.getMessageType() != MessageType.END_OF_CONVERSATION) && (voiceInput.getMessageType() != MessageType.STOP)) {
-      SimpleCard card = new SimpleCard();
-      card.setContent(builder.getRawText());
-      card.setTitle("Alexa + DERP Wizard");
-
-      @SuppressWarnings("unchecked")
-      VoiceOutput<SpeechletResponse> voiceOutput = (VoiceOutput<SpeechletResponse>) VoiceMessageFactory.buildOutputMessage(builder.build(), InterfaceType.ALEXA);
-      SpeechletResponse speechletResponse = (SpeechletResponse) voiceOutput.getImplInstance();
-      speechletResponse.setCard(card);
-
-      responseEnvelope.setResponse(speechletResponse);
+      @HeaderParam("Signature") String signature, @QueryParam("testFlag") Boolean testFlag){
+
+    ObjectMapper mapper = new ObjectMapper();
+    CommonMetadata metadata = null;
+    try{
+      if (request.getRequest() == null) {
+        throw new DerpwizardException(DerpwizardExceptionReasons.MISSING_INFO.getSsml(),"Missing request body."); //TODO: create AlexaException
+      }
+      
+      if(testFlag == null || testFlag == false){ 
+        AlexaUtils.validateAlexaRequest(request, signatureCertChainUrl, signature);
+      }
+      
+      Map<String, Object> sessionAttributes = request.getSession().getAttributes();
+      metadata = mapper.convertValue(sessionAttributes, new TypeReference<CommonMetadata>(){});
+      SsmlDocumentBuilder builder = new SsmlDocumentBuilder(UNSUPPORTED_SSML_TAGS);
+      VoiceInput voiceInput = VoiceMessageFactory.buildInputMessage(request.getRequest(), InterfaceType.ALEXA);
+      manager.handleRequest(voiceInput, builder);
+  
+      SpeechletResponseEnvelope responseEnvelope = new SpeechletResponseEnvelope();
+      responseEnvelope.setSessionAttributes(request.getSession().getAttributes());
+  
+      // Create a VoiceOutput object with the SSML content generated by the manater
+      if ((voiceInput.getMessageType() != MessageType.END_OF_CONVERSATION) && (voiceInput.getMessageType() != MessageType.STOP)) {
+        SimpleCard card = new SimpleCard();
+        card.setContent(builder.getRawText());
+        card.setTitle("Alexa + DERP Wizard");
+  
+        @SuppressWarnings("unchecked")
+        VoiceOutput<SpeechletResponse> voiceOutput = (VoiceOutput<SpeechletResponse>) VoiceMessageFactory.buildOutputMessage(builder.build(), InterfaceType.ALEXA);
+        SpeechletResponse speechletResponse = (SpeechletResponse) voiceOutput.getImplInstance();
+        speechletResponse.setCard(card);
+  
+        responseEnvelope.setResponse(speechletResponse);
+      }
+  
```

Comment:
Did we want this bit from Quip?:
else{
      outputSpeech = null;
      card = null;
      speechletResponse.setShouldEndSession(true);
    }